lintro/ai/review/chunker/workflow_scripts.py

```
@@ -0,0 +1,1013 @@
+                    cwd=branch_cwd,
+                )
+    return False
+
+
+def _shell_cwd_after_line(*, line: str, cwd: str) -> str:
+    """Return cwd after sequential ``cd`` commands that persist in the shell."""
+    if "||" in line or "|" in line:
+        return _normalize_posix_shell_path(path=cwd)
+    current = _normalize_posix_shell_path(path=cwd)
+    for segment in re.split(r"&&|;", line):
+        segment = segment.strip()
```

Ruff (0.15.18)

[warning] `for` loop variable `segment` overwritten by assignment target

(PLW2901)